basictracer/tracer.py

```diff
@@ -26,6 +27,7 @@ def __init__(self, recorder=None, sampler=None):
         super(BasicTracer, self).__init__()
         self.recorder = NoopRecorder() if recorder is None else recorder
         self.sampler = DefaultSampler(1) if sampler is None else sampler
+        self._active_span_source = ThreadLocalActiveSpanSource()
```

Comment:
One thing is that maybe we should let the users specify their own active span source at ctor time, and default to the thread local one, so they can do something like:

`BasicTracer(source=GeventActiveSpanSource())`